lib/ext/heroku/command/addons.rb

```diff
@@ -219,7 +219,7 @@ def add
       when 422 # add-on resource not found, should probably be 404
```

Comment:
Yeah I think I have comments on the API implementation to the effect of `422? 404? argh`.

I land on the `422` case because while one of the args is resulting in a lookup that can't be met, the *route* and action isn't missing so a 404 has incorrect implied semantics to the user agent. The only thing wrong with the request is the args provided, not the semantics.

Thoughts?